### PR TITLE
Prevent tests from running on forks

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

This is to prevent consumption of LFS resources. 

### :pushpin: Resources

Examples, notebooks, and links to useful references.
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
https://github.com/atharva-2001/tardis/actions/runs/4742893534

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by andrewfullard** [comment]
I take it forks cannot use caches in the same way? And even if they can, they will still have to generate a cache to begin with?

**Review by andrewfullard — DISMISSED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ defaults:
 
 jobs:
   build:
-
+    if: github.repository_owner=='tardis-sn'
```
Comment (epassaro) [inline .github/workflows/tests.yml:32]: Leave spaces before and after ==

**Comment by atharva-2001** [comment]
https://github.com/tardis-sn/tardis/blob/b3bc761aa9c7d9a185b2d76eb504048c1e71e1de/.github/workflows/tests.yml#L22-L24

I'm afraid I don't know if this works completely as intended.
The workflow should run only once for a commit, right?
`${{ github.ref }}` is different for forks and for upstream repo.
https://github.com/atharva-2001/tardis/actions/runs/4798029406/jobs/8535766612
https://github.com/tardis-sn/tardis/actions/runs/4798029889/jobs/8535760860

Here is the workflow file.
https://github.com/tardis-sn/tardis/actions/runs/4798029889/workflow

**Comment by epassaro** [comment]
> https://github.com/tardis-sn/tardis/blob/b3bc761aa9c7d9a185b2d76eb504048c1e71e1de/.github/workflows/tests.yml#L22-L24
> 
> I'm afraid I don't know if this works completely as intended. The workflow should run only once for a commit, right? `${{ github.ref }}` is different for forks and for upstream repo. https://github.com/atharva-2001/tardis/actions/runs/4798029406/jobs/8535766612 https://github.com/tardis-sn/tardis/actions/runs/4798029889/jobs/8535760860
> 
> Here is the workflow file. https://github.com/tardis-sn/tardis/actions/runs/4798029889/workflow

As far as I remember, that creates a group depending if the workflow runs on a branch of the repository or runs on a pull request. 

The part ${{ github.head_ref || github.ref }} ` means, if `github.head_ref` does not exist fallback to `github.ref`. One of these variables exist if the workflow runs in a branch of the repo, the other exist in pull requests (or both scenarios?, I don't remember).

It wasn't designed with forks in mind, I don't know what happens there.

**Comment by atharva-2001** [comment]
I agree- this is working correctly. My apologies. I also found this- might be of interest:
https://stackoverflow.com/a/75403978/11974464

File: .github/workflows/tests.yml (hunk 1 above)
Comment (atharva-2001) [reply .github/workflows/tests.yml]: Done via https://github.com/tardis-sn/tardis/pull/2280/commits/8351415447c05920aa85e0d657e4ba811dc20528

**Review by andrewfullard — APPROVED** [review]

**Review by ymatsumu — APPROVED** [review]

**Pull request merged**